docs/privatelink.md

```diff
@@ -0,0 +1,276 @@
+publicly availble on the Internet. The OpenShift Installer allows creating
+clusters that have their services published only on the internal network by
+setting `publish: Internal` in the install-config.yaml.
+
+Since Hive is usually running outside the network where the cluster is being
+deployed, special consideration must be taken to ensure it has access to the
+clsuter's API server. While there are multiple ways to create publicly
+accessible endpoints that are limited such that only Hive can access them,
+these are generally not acceptable by security focused customers.
+
+Cloud providers usually have a method that allows accessing private services
+from another network using the provider's internal networking instead of going
+over the public Internet. The PrivateLink controller leverages these features
+to enable Hive to access the APIs of these private cluster without eposing them
+to the Internet.
+
+## Architecture
+
+The PrivateLink controller has been designed to enable a hive cluster to exist
+on one platform while it deploys private clusters on any of the supported
+platforms. It accomplishes this by breaking the functionality into two
+segments (actuators):
+
+### Hub Actuator
+
+This actuator creates the necessary resources on the cloud provider that Hive is
+running on. Its job is to provide a DNS record, and any other necessary resources,
+to enable the Hive cluster to resolve the cluster's API to the Endpoint created
+by the Link Actuator.
+
+Supported Platforms: [AWS](#aws-hub-actuator)
+
+### Link Actuator
+
+This actuator creates the necessary resources on the cloud provider that the
+private cluster is being provisioned into. Its job is to create an Endpoint,
+and any other necessary resources, to enable the Hive cluster to connect to the
+cluster's API without enabling public access.
+
+Supported Platforms: [GCP](#gcp-link-actuator)
+
+Note: Support for AWS private clusters is still managed by the origional [AWS
+PrivateLink Controller](awsprivatelink.md). The goal is to eventually merge its
+functionality into this PrivateLink Controller.
+
+## AWS Hub Actuator Configuration
+
+This actuator is used when the Hive cluster exists in the AWS platform. It will
+ensure a new DNS zone exists for the private cluster and create a DNS record
+which resolves to the Endpoint created by the Link Actuator. This zone will
+then be attached to the associatedVPCs configured in the hiveconfig.
+
+Note: The AWS Hub Actuator currently uses the same HiveConfig parameters as the
+origional AWS PrivateLink controller. This is to avoid configuration drift
+between the two controllers until they can be merged together.
+
+### Configure the primary AWS credentailsSecretRef
+
+The Hub Actuator needs an AWS Service Account in order to manage the DNS
+Zone and records. The credentials for this service account should be stored in
+a secret in the Hive namespace. This secret can then be configured via the
```

Comment:
nit: it's the namespace denoted by `HiveConfig.spec.targetNamespace`, which _defaults_ to `hive`. This is worthy of a footnote at most, if we want to bother with it at all -- thoughts?

One other instance on [L121 in this rev](https://github.com/openshift/hive/pull/2480/files#diff-108c0ef968036977db06b3548a995872daef4ac5d5e0002b3972106c02a7590dR121).